### PR TITLE
Missing cosmosDenom when building the prophecy id

### PR DESCRIPTION
**Pull request by juniuszhou**
End to end test passed.

### CODE REVIEW

**Comment by Brando753** [comment]
Heads up @juniuszhou This looks good but I want to add some tests as well and remove unnecessary fields from the claimdata such as symbol, name, and decimals before we merge this in.

File: x/ethbridge/types/msgs.go [hunk 1]
```diff
@@ -207,29 +242,45 @@ func ComputeProphecyID(cosmosSender string, sequence uint64, ethereumReceiver st
...
+			Type: uint8Ty,
+		},
+		{
+			Type: int32Ty,
+		},
+		{
+			Type: boolTy,
+		},
+		{
+			Type: uint256Ty,
```
Comment (Brando753) [inline x/ethbridge/types/msgs.go:264]: the claim data struct in solidity labels nonce as a uint128 why did you change this to uint256?

**Review by Brando753 — CHANGES_REQUESTED** [review]
Once I understand this one comment change I will merge.

File: x/ethbridge/types/msgs.go (hunk 1 above)
Comment (juniuszhou) [reply x/ethbridge/types/msgs.go:264]: there is no uint128 type in go, so I use the big.Int as big.NewInt(int64(globalNonce)). In abi argument, we need a uint256 to pack it.

File: x/ethbridge/types/msgs.go [hunk 2]
```diff
@@ -79,16 +79,28 @@ func (msg MsgLock) GetSigners() []sdk.AccAddress {
 }
 
 // GetProphecyID get prophecy ID for lock message
-func (msg MsgLock) GetProphecyID(doublePeggy bool, sequence, globalNonce uint64, tokenAddress string) []byte {
+func (msg MsgLock) GetProphecyID(
```
Comment (smartyalgo) [inline x/ethbridge/types/msgs.go:82]: Not required for this PR.

Have we decided against merging these two functions?

Comment (juniuszhou) [reply x/ethbridge/types/msgs.go:82]: you means GetProphecyID for MsgLock and MsgBurn?

File: smart-contracts/contracts/CosmosBridge.sol [hunk 3]
```diff
@@ -132,12 +132,20 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
    * @param _networkDescriptor Unique identifier of the network
```
Comment (smartyalgo) [inline smart-contracts/contracts/CosmosBridge.sol:132]: Comments outdated

File: x/ethbridge/types/msgs.go (hunk 2 above)
Comment (Brando753) [reply x/ethbridge/types/msgs.go:82]: without the generics support, let's just leave it as is.

**Review by Brando753 — APPROVED** [review]

**Review by kleimanb81 — COMMENTED** [review]
Nice job, quick response!
